src/components/ScmRepositoryCreateEditPage/ScmRepositoryCreateEditPage.tsx

```diff
@@ -0,0 +1,268 @@
+        navigate(`/scm-repositories/${id}`);
+      })
+      .catch(() => {
+        throw new Error('Failed to edit SCM Repository.');
+      });
+  };
+
+  const { fields, onChange, reinitialize, onSubmit, isSubmitDisabled } = useForm(
+    isEditPage ? editFormConfig : createFormConfig,
+    isEditPage ? submitUpdate : submitCreate
+  );
+
+  useEffect(() => {
+    if (isEditPage) {
+      if (urlPathParams.scmRepositoryId) {
+        serviceContainerEditPageGetRunner({ serviceData: { id: urlPathParams.scmRepositoryId } }).then((response: any) => {
+          const scmRepository: SCMRepository = response.data;
+
+          setScmRepository(scmRepository);
+          setId(scmRepository.id);
+          reinitialize({
+            externalUrl: scmRepository.externalUrl,
+            // preBuildSyncEnabled: scmRepository.preBuildSyncEnabled,
+          });
+        });
+      } else {
+        throw new Error(`Invalid scmRepositoryId: ${urlPathParams.scmRepositoryId}`);
+      }
+    }
+  }, [isEditPage, urlPathParams.scmRepositoryId, serviceContainerEditPageGetRunner, reinitialize]);
+
+  const formComponent = (
+    <ContentBox padding>
+      <div className="w-70">
+        <Form
+          onSubmit={(e) => {
+            e.preventDefault();
+          }}
+        >
+          {!isEditPage && (
+            <FormGroup
+              isRequired
+              label="SCM Repository URL"
+              fieldId="scmUrl"
+              labelIcon={
+                <TooltipWrapper tooltip="The URL of the SCM to be created, this can be either an internal or external URL." />
+              }
+              helperText={
+                <FormHelperText isHidden={fields.scmUrl.state !== 'error'} isError>
+                  {fields.scmUrl.errorMessages?.join(' ')}
+                </FormHelperText>
+              }
+            >
+              <TextInput
+                isRequired
+                validated={fields.scmUrl.state}
+                type="text"
+                id="scmUrl"
+                name="scmUrl"
+                value={fields.scmUrl.value}
+                autoComplete="off"
+                onChange={(scmUrl) => {
+                  onChange('scmUrl', scmUrl);
+                }}
+              />
+            </FormGroup>
+          )}
+          {isEditPage && (
+            <>
+              <FormGroup
+                label="Internal SCM URL"
+                fieldId="internalUrl"
+                labelIcon={
+                  <TooltipWrapper tooltip="URL to the internal SCM repository, which is the main repository used for the builds." />
+                }
+              >
+                {scmRepository && <ScmRepositoryUrl internalScmRepository={scmRepository} />}
+              </FormGroup>
+              <FormGroup
+                label="External SCM URL"
+                fieldId="externalUrl"
+                labelIcon={<TooltipWrapper tooltip="URL to the upstream SCM repository." />}
+              >
```

Comment:
This FormGroup misses the `helperText` and `isRequired`

Reply:
HelperText added. External SCM URL can be empty when you editing the scm repo.